src/mpyl/projects/versioning.py

```diff
@@ -0,0 +1,164 @@
+"""Versioning and upgrade utilities for mpyl projects.
+### Writing an upgrade script
+
+To write an upgrade script, create a new class that inherits from `Upgrader` and
+implements the `upgrade` method. This class should then be added to the `UPGRADERS`
+list in this module.
+
+"""
+import pkgutil
+from io import StringIO
+from pathlib import Path
+from typing import Optional, Generator
+import copy
+
+from deepdiff import DeepDiff
+from ruamel.yaml import YAML
+from ruamel.yaml.compat import ordereddict
+
+VERSION_FIELD = "mpylVersion"
+
+
+def get_releases() -> list[str]:
+    embedded_releases = pkgutil.get_data(__name__, "releases/releases.txt")
+    if not embedded_releases:
+        raise ValueError("File releases/releases.txt not found in package")
+    releases = embedded_releases.decode("utf-8").strip().splitlines()
+    return list(reversed(releases))
+
+
+def get_latest_release() -> str:
+    return get_releases()[0]
+
+
+def yaml_to_string(serializable: object, yaml: YAML) -> str:
```

Comment:
Could be part of the utils package

Reply:
[[TECH-556] Move yaml utility methods to their own package](https://github.com/Vandebron/mpyl/pull/207/commits/4beb088629384d8d28c4e2773a997f0eee0f2d10)